src/cdk/tree/tree.spec.ts

```diff
@@ -1015,11 +1060,15 @@ class FakeDataSource extends DataSource<TestData> {
   get data() { return this._dataChange.getValue(); }
   set data(data: TestData[]) { this._dataChange.next(data); }
 
-  constructor(public treeControl: TreeControl<TestData>) {
+  constructor(public treeControl: TreeControl<TestData>, addChildren = false) {
     super();
     for (let i = 0; i < 3; i++) {
       this.addData();
     }
+    if (addChildren) {
+      const parent = this.data[0];
+      this.addChild(parent, true);
+    }
```

Comment:
I added this `addChildren` flag as a way to optionally add children to the data source before the tree is rendered, because aria attribute changes were not reflected in the DOM otherwise. Ideally I would remove this. See other comment -- I need to find a way to make the parent node update its data when a child is added to it. I think this might just be a test change, but I could use another pair of eyes on this.

Reply:
I think this is probably the same issue as I commented on in https://github.com/angular/components/pull/17818